src/main/java/com/metamx/metrics/JvmMonitor.java

```diff
@@ -156,12 +53,21 @@ public JvmMonitor(Map<String, String[]> dimensions)
   @Override
   public boolean doMonitor(ServiceEmitter emitter)
   {
+    emitJvmMemMetrics(emitter);
+    emitDirectMemMetrics(emitter);
+    emitGcMetrics(emitter);
+
+    return true;
+  }
+
+  // These metrics are going to be replaced by new jvm/gc/mem/* metrics
+  @Deprecated
+  private void emitJvmMemMetrics(ServiceEmitter emitter)
+  {
     // I have no idea why, but jvm/mem is slightly more than the sum of jvm/pool. Let's just include
     // them both.
-
-    // jvm/mem
     final Map<String, MemoryUsage> usages = ImmutableMap.of(
-        "heap",    ManagementFactory.getMemoryMXBean().getHeapMemoryUsage(),
+        "heap", ManagementFactory.getMemoryMXBean().getHeapMemoryUsage(),
```

Comment:
Maybe cache `ManagementFactory.getMemoryMXBean()` in a field? This is a call of `synchronized` method. It would also be consistent with jStat-based metrics, because counters are cached and all preparations are done on JvmMonitor initialization, doing only minimal work during `doMonitor()` calls.